src/ostree/ot-admin-builtin-upgrade.c

```diff
@@ -104,6 +121,21 @@ ot_admin_builtin_upgrade (int argc, char **argv, GCancellable *cancellable, GErr
         }
     }
 
+  /* In the pull-only case, we do a cleanup here to ensure that if multiple
+   * commits were pulled, we garbage collect any intermediate data before
+   * pulling more.  This is really a best practice in general, but for maximum
+   * compatiblity, we only do cleanup if a user specifies the new --pull-only
+   * option.  Otherwise, we would break the case of trying to deploy a commit
+   * that isn't directly referenced.
+   */
```

Comment:
I'm still not sure I understand this. Is there any risk to leaving the intermediate data until the cleanup that happens after deployment? Isn't this making us lose potential static delta jumps between consecutive `--pull-only` operations?